lib/tls.js

```diff
@@ -199,34 +199,35 @@ exports.checkServerIdentity = function checkServerIdentity(hostname, cert) {
   let valid = false;
   let reason = 'Unknown reason';
 
+  const hasAltNames =
+    dnsNames.length > 0 || ips.length > 0 || uriNames.length > 0;
+
+  hostname = unfqdn(hostname);  // Remove trailing dot for error messages.
+  const hostParts = splitHost(hostname);
+  const wildcard = (pattern) => check(hostParts, pattern, true);
+  const noWildcard = (pattern) => check(hostParts, pattern, false);
```

Comment:
It would be better to move these back inside an `if` statement.

`unfqdn()` and `splitHost()` are somewhat expensive operations that the user shouldn't have to pay for when the other branches are taken.